src/helpers/copyContract.js

```diff
@@ -0,0 +1,26 @@
+export default async function copyContract(contractClass, address, txParams = {}) {
+  address = address.replace('0x', '')
+
+  // This is EVM assembly will return of the code of a foreign address.
+  //
+  // operation    | bytecode   | stack representation
+  // =================================================
+  // push20 ADDR  | 0x73 ADDR  | ADDR
+  // dup1         | 0x80       | ADDR ADDR
+  // extcodesize  | 0x3B       | ADDR 0xCS
+  // dup1         | 0x80       | ADDR 0xCS 0xCS
+  // swap2        | 0x91       | 0xCS 0xCS ADDR
+  // push1 00     | 0x60 0x00  | 0xCS 0xCS ADDR 0x00
+  // dup1         | 0x80       | 0xCS 0xCS ADDR 0x00 0x00
+  // swap2        | 0x91       | 0xCS 0xCS 0x00 0x00 ADDR
+  // extcodecopy  | 0x3C       | 0xCS
+  // push1 00     | 0x60 0x00  | 0xCS 0x00
+  // return       | 0xF3       |
+
+  const ASM_CODE_COPY = `0x73${address}803b8091600080913c6000f3`
+
+  const params = Object.assign({}, contractClass.defaults(), txParams, { to: 0x0, data: ASM_CODE_COPY })
+  const txHash = web3.eth.sendTransaction(params)
+  const receipt = web3.eth.getTransactionReceipt(txHash)
```

Comment:
Web3's `getTransactionReceipt` blocking version will return once the request is sent to the node, **not** once the transaction is mined. This method will always return the receipt in ganache (since it mines almost instantly), but in geth or parity will return `null` until the tx is actually mined.

We actually need to add a loop that checks every X time for the tx to be mined, which is what truffle ends up doing for regular transactions ([see here](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-contract/contract.js#L140)), or use web3 `checkForContractAddress`, which does something similar but block-based instead of time-based ([see here](https://github.com/ethereum/web3.js/blob/develop/lib/web3/contract.js#L96)).